vm.c

```diff
@@ -2995,6 +3055,16 @@ ruby_vm_destruct(rb_vm_t *vm)
         }
         RB_ALTSTACK_FREE(vm->main_altstack);
         if (objspace) {
+            if (rb_free_on_shutdown) {
+                rb_objspace_free_objects(objspace);
+                rb_free_generic_iv_tbl_();
+                if (th) {
+                    xfree(nt);
```

Comment:
on Mac, this fails with,
```
pointer being freed was not allocated
```

but I am confused since it seems like,

```
vm->ractor.main_thread
```

is always `ZALLOC`ed in `Init_BareVM`.